From 852b2e684c5afa09d53a2909afe3d8058fbda901 Mon Sep 17 00:00:00 2001
From: Zach Brown <zach.brown@destroystokyo.com>
Date: Wed, 4 Apr 2018 21:00:43 -0400
Subject: [PATCH] MC-124833

Conflicting reports of what software this bug does and does not affect.
Given the trivial impact of fixing it and an overabundance of caution,
we'll patch it in Paper as well. This will probably be a short lived
patch, with it being fixed in Vanilla proper 1.13

diff --git a/src/main/java/net/minecraft/server/EntityItemFrame.java b/src/main/java/net/minecraft/server/EntityItemFrame.java
index 10085675e..b2eabfce1 100644
--- a/src/main/java/net/minecraft/server/EntityItemFrame.java
+++ b/src/main/java/net/minecraft/server/EntityItemFrame.java
@@ -92,6 +92,10 @@ public class EntityItemFrame extends EntityHanging {
             }
 
             itemstack.a((EntityItemFrame) null);
+
+            // Paper - MC-124833 - conflicting reports of what server software this does and doesn't affect.
+            // It's a one liner with near-zero impact so we'll patch it anyway just in case
+            this.setItem(null); // OBFHELPER - ItemStack.EMPTY
         }
     }
 
-- 
2.40.0.windows.1

